dwave/preprocessing/presolve/cypresolve.pyx

```diff
@@ -196,11 +197,17 @@ cdef class cyPresolver:
 
         return changes
 
-    cpdef bint presolve(self) except*:
+    cpdef bint presolve(self, double time_limit = float("inf")) except*:
         """Apply any loaded presolve techniques to the held constrained quadratic model.
 
         Must be called after :meth:`normalize`.
 
+        Args:
+            time_limit:
+                A time limit in seconds.
+                The presolve rounds will terminate after the time limit is exceeded.
+                Defaults to ``float("inf")``.
```

Comment:
May want to add detail that it will terminate after (periodically) checking if the time limit has been exceeded.

Reply:
I think I'll leave this as is for the moment, though everything needs a doc facelift at some point.